vectorstores/pgvector/pgvector_test.go

```diff
@@ -0,0 +1,409 @@
+package pgvector_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/tmc/langchaingo/chains"
+	"github.com/tmc/langchaingo/embeddings"
+	"github.com/tmc/langchaingo/llms/openai"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/vectorstores"
+	"github.com/tmc/langchaingo/vectorstores/pgvector"
+)
+
+func preCheckEnvSetting(t *testing.T) {
+	t.Helper()
+
+	pgvectorURL := os.Getenv("PGVECTOR_CONNECTION_STRING")
+	if pgvectorURL == "" {
+		t.Skip("Must set PGVECTOR_CONNECTION_STRING to run test")
+	}
+
+	if openaiKey := os.Getenv("OPENAI_API_KEY"); openaiKey == "" {
+		t.Skip("OPENAI_API_KEY not set")
+	}
+}
+
+func getTestCollectionName() string {
+	return fmt.Sprintf("test-collection-%s", uuid.New().String())
```

Comment:
Nit: naming this `makeNewCollectionName` will make it clearer that it generates a new random name every time it's called, rather than "getting" some pre-set name

Reply:
Good Point.